ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -88,8 +88,24 @@ const STACK_DIRECTION: StackDirection = StackDirection::GrowsDown;
 #[inline(never)]
 pub extern "C" fn __yk_break() {}
 
+/// A simple front end for the X86_64 code generator.
+pub(crate) struct X64CodeGenFrontEnd;
+
+impl CodeGen for X64CodeGenFrontEnd {
+    fn codegen(&self, m: Module) -> Result<Arc<dyn CompiledTrace>, CompilationError> {
+        let ra = Box::new(SpillAllocator::new(StackDirection::GrowsDown));
+        X64CodeGen::new(&m, ra)?.codegen()
```

Comment:
Actually I'm not sure what this change does. I thought you were going to cache the code generator so that you don't have to reinstantiate it every time, but it looks like we are doing that anyway here?

Is this change just setting us up to do that in a future PR, or?

Reply:
Yes, this is mostly setup: the point is that the `CodeGen` trait is now "correct" but this PR only minimally changes the x86_64 backend.

Reply:
So just for clarification: this PR just makes a place for shared caches to live in the `Frontend` struct?

Reply:
I tend to to think of it as "the `CodeGen` trait no longer prevents caching", but ultimately it's the same thing, yes.